packages/base/src/Boot.ts

```diff
@@ -27,6 +27,10 @@ const boot = async (): Promise<void> => {
 	}
 
 	const bootExecutor = async (resolve: PromiseResolve) => {
+		if (typeof window === "undefined") {
```

Comment:
I checked that `document` is a preferred way for detecting client-side, since window is implemented in some server environments like Deno

`typeof document === "undefined"`

from:
https://remix.run/docs/en/v1/guides/constraints#document-guard

Reply:
good point, done.